pkg/testsuite/mock/wallet_transactions.go

```diff
@@ -347,20 +347,66 @@ func (w *Wallet) CreateImplicitAccountFromInput(transactionName string, inputNam
+// CreateImplicitAccountFromInput creates an implicit account output.
+func (w *Wallet) CreateImplicitAccountAndBasicOutputFromInput(transactionName string, inputName string, recipientWallet *Wallet) *iotago.SignedTransaction {
 	input := w.Output(inputName)
-	implicitAccountID := iotago.AccountIDFromOutputID(input.OutputID())
 
-	basicOutput, isBasic := input.Output().(*iotago.BasicOutput)
-	if !isBasic {
-		panic(fmt.Sprintf("output with alias %s is not *iotago.BasicOutput", inputName))
+	implicitAccountOutput := &iotago.BasicOutput{
+		Amount: MinIssuerAccountAmount(w.Node.Protocol.CommittedAPI().ProtocolParameters()),
+		Mana:   AccountConversionManaCost(w.Node.Protocol.CommittedAPI().ProtocolParameters()),
+		UnlockConditions: iotago.BasicOutputUnlockConditions{
+			&iotago.AddressUnlockCondition{Address: recipientWallet.ImplicitAccountCreationAddress()},
+		},
+		Features: iotago.BasicOutputFeatures{},
 	}
-	if basicOutput.UnlockConditionSet().Address().Address.Type() != iotago.AddressImplicitAccountCreation {
-		panic(fmt.Sprintf("output with alias %s is not an implicit account", inputName))
+
+	remainderBasicOutput := &iotago.BasicOutput{
+		Amount: input.BaseTokenAmount() - MinIssuerAccountAmount(w.Node.Protocol.CommittedAPI().ProtocolParameters()),
+		Mana:   input.StoredMana() - AccountConversionManaCost(w.Node.Protocol.CommittedAPI().ProtocolParameters()),
+		UnlockConditions: iotago.BasicOutputUnlockConditions{
+			&iotago.AddressUnlockCondition{Address: recipientWallet.Address()},
+		},
+		Features: iotago.BasicOutputFeatures{},
 	}
 
+	signedTransaction := w.createSignedTransactionWithOptions(
+		transactionName,
+		[]uint32{0},
+		WithInputs(input),
+		WithOutputs(implicitAccountOutput, remainderBasicOutput),
+	)
+
+	// register the outputs in the recipient wallet (so wallet doesn't have to scan for outputs on its addresses)
+	recipientWallet.registerOutputs(transactionName, signedTransaction.Transaction)
+
+	// register the implicit account as a block issuer in the wallet
+	implicitAccountID := iotago.AccountIDFromOutputID(recipientWallet.Output(fmt.Sprintf("%s:0", transactionName)).OutputID())
+	recipientWallet.SetBlockIssuer(implicitAccountID)
+
+	return signedTransaction
+}
+
+func (w *Wallet) TransitionImplicitAccountToAccountOutput(transactionName string, inputNames []string, opts ...options.Option[builder.AccountOutputBuilder]) *iotago.SignedTransaction {
+	var implicitAccountOutput *utxoledger.Output
+	inputs := make(utxoledger.Outputs, 0, len(inputNames))
+	for _, inputName := range inputNames {
+		input := w.Output(inputName)
+		basicOutput, isBasic := input.Output().(*iotago.BasicOutput)
+		if !isBasic {
+			panic(fmt.Sprintf("output with alias %s is not *iotago.BasicOutput", inputName))
+		}
+		if basicOutput.UnlockConditionSet().Address().Address.Type() == iotago.AddressImplicitAccountCreation {
+			implicitAccountOutput = input
```

Comment:
You should check if `implicitAccountOutput` was already set, and if so, you should panic because there was already another implicit account.